packages/opentelemetry-tracing/src/BasicTracerProvider.ts

```diff
@@ -99,4 +102,8 @@ export class BasicTracerProvider implements api.TracerProvider {
       api.propagation.setGlobalPropagator(config.propagator);
     }
   }
+
+  shutdown(): void {
```

Comment:
no callback? not waiting for the shutdown to finish?

Reply:
I had implemented all of this assuming users would not be calling shutdown manually. I can add a callback parameter if you think that would be a good feature.

Reply:
I do

Reply:
What do you mean by "not waiting for the shutdown to finish"? I assumed shutdown was a blocking function. I added a callback option which will be passed to the shutdown function in the processor but do I need to make this async and add an await or should that be enough?

Reply:
Issue with adding the callback option in these shutdown methods: any function used as a signal handler cannot have any args so adding callbacks is a bit tricky. We can get around this by adding @ts-ignore to ignore strict function typing but this may be ill advised? I compiled it using this and the test cases worked fine. Another solution I could try would be to have a private onShutdown function which is bound to the signal handler and a public shutdown function which accepts a callback - they would have identical functionality aside from the callback arg but this may look confusing as it requires duplicated code (I wish typescript had overloaded functions!)

Reply:
I am strongly against ts-ignore

Reply:
I ended up not using @ts-ignore, I also added callbacks and make the meter provider shutdown blocking.